File: leetcode-algorithms/080. Remove Duplicates from Sorted Array II/solution.py
from typing import List
def removeDuplicates(nums: List[int]) -> int:
    if len(nums) == 0:
        return 0
    if len(nums) <= 2:
        return len(nums)
    i,t = 0,0
    count = 0
    while i < len(nums):
        if count < 2:
            count +=1
            nums[t] = nums[i]
            t += 1
        if i<len(nums)-1 and nums[i] != nums[i+1]:
            count = 0
        i+=1

    return t

l = [0,0,1,1,1,1,2,3,3]
print(removeDuplicates(l))
print(l)

File: leetcode-algorithms/080. Remove Duplicates from Sorted Array II/solution.jl
function removeDuplicates(nums::Array)::Int
    if length(nums) == 0
        return 0
    end
    if length(nums) <= 2
        return length(nums)
    end
    i,t = 1,1
    count = 0
    while i <= length(nums)
        if count < 2
            count +=1
            nums[t] = nums[i]
            t += 1
        end
        if i<=length(nums)-1 && nums[i] != nums[i+1]
            count = 0
        end
        i+=1
    end
    return t-1
end;

l = [0,0,1,1,1,1,2,3,3]
println(removeDuplicates(l))
println(l)
